File: ios/Sources/ILDSDesignSystem/IldsButton.swift
import ILDSTokens
import SwiftUI

// Figma component set 13472:2804 — mirrors lib/ilds_button.dart + web Button.tsx.
// TODO(web/desktop): hover states — mobile-first defer (matches Flutter).

public enum IldsButtonType: Sendable {
    case primary, secondary, tertiary
}

public enum IldsButtonSize: Sendable {
    case large, medium, small
}

public enum IldsButtonAppearance: Sendable {
    case normal, destructive
}

public struct IldsButton: View {
    private let label: String?
    private let action: () -> Void
    private let type: IldsButtonType
    private let size: IldsButtonSize
    private let appearance: IldsButtonAppearance
    private let isDisabled: Bool
    private let isLoading: Bool
    private let iconOnly: Bool
    private let semanticLabel: String?
    private let leading: AnyView?
    private let trailing: AnyView?
    private let icon: AnyView?

    /// Labeled button with optional leading/trailing icon slots.
    public init(
        _ label: String,
        action: @escaping () -> Void,
        type: IldsButtonType = .primary,
        size: IldsButtonSize = .large,
        appearance: IldsButtonAppearance = .normal,
        isDisabled: Bool = false,
        isLoading: Bool = false,
        leading: (any View)? = nil,
        trailing: (any View)? = nil
    ) {
        self.label = label
        self.action = action
        self.type = type
        self.size = size
        self.appearance = appearance
        self.isDisabled = isDisabled
        self.isLoading = isLoading
        self.iconOnly = false
        self.semanticLabel = nil
        self.leading = leading.map { AnyView($0) }
        self.trailing = trailing.map { AnyView($0) }
        self.icon = nil
    }

    /// Icon-only button (Figma 13472:2810 L, 13472:3718 S). Requires semanticLabel.
    public init(
        icon: some View,
        semanticLabel: String,
        action: @escaping () -> Void,
        type: IldsButtonType = .primary,
        size: IldsButtonSize = .large,
        appearance: IldsButtonAppearance = .normal,
        isDisabled: Bool = false,
        isLoading: Bool = false
    ) {
        self.label = nil
        self.action = action
        self.type = type
        self.size = size
        self.appearance = appearance
        self.isDisabled = isDisabled
        self.isLoading = isLoading
        self.iconOnly = true
        self.semanticLabel = semanticLabel
        self.leading = nil
        self.trailing = nil
        self.icon = AnyView(icon)
    }

    public var body: some View {
        Button(action: action) {
            IldsButtonLabel(
                label: label,
                type: type,
                size: size,
                appearance: appearance,
                isDisabled: isDisabled,
                isLoading: isLoading,
                iconOnly: iconOnly,
                leading: leading,
                trailing: trailing,
                icon: icon
            )
        }
        .buttonStyle(
            IldsButtonStyle(
                type: type,
                size: size,
                appearance: appearance,
                isDisabled: isDisabled,
                isLoading: isLoading,
                iconOnly: iconOnly
            )
        )
        .disabled(isDisabled || isLoading)
        .accessibilityLabel(iconOnly ? (semanticLabel ?? "") : (label ?? ""))
    }
}

// MARK: - Label row

private struct IldsButtonLabel: View {
    let label: String?
    let type: IldsButtonType
    let size: IldsButtonSize
    let appearance: IldsButtonAppearance
    let isDisabled: Bool
    let isLoading: Bool
    let iconOnly: Bool
    let leading: AnyView?
    let trailing: AnyView?
    let icon: AnyView?

    var body: some View {
        let colors = IldsButtonColors.resolve(
            type: type,
            appearance: appearance,
            isPressed: false,
            isDisabled: isDisabled,
            isLoading: isLoading
        )
        let metrics = IldsButtonMetrics(type: type, size: size, iconOnly: iconOnly)
        let leadingContent = iconOnly ? icon : leading

        HStack(spacing: metrics.gap) {
            if let leadingContent {
                iconSlot(leadingContent, color: colors.foreground, size: metrics.iconSlot)
                if !iconOnly, label != nil {
                    Spacer().frame(width: 0)
                }
            }
            if !iconOnly, let label {
                Text(label)
                    .font(metrics.font)
                    .foregroundStyle(colors.foreground)
                    .lineLimit(1)
            }
            if isLoading {
                ProgressView()
                    .progressViewStyle(.circular)
                    .tint(colors.foreground)
                    .frame(width: metrics.iconSlot, height: metrics.iconSlot)
            } else if !iconOnly, let trailing {
                iconSlot(trailing, color: colors.foreground, size: metrics.iconSlot)
            }
        }
        .frame(minHeight: metrics.minHeight)
        .padding(metrics.padding)
    }

    private func iconSlot(_ content: AnyView, color: Color, size: CGFloat) -> some View {
        content
            .foregroundStyle(color)
            .frame(width: size, height: size)
    }
}

// MARK: - Button style (pressed / overlay)

private struct IldsButtonStyle: ButtonStyle {
    let type: IldsButtonType
    let size: IldsButtonSize
    let appearance: IldsButtonAppearance
    let isDisabled: Bool
    let isLoading: Bool
    let iconOnly: Bool

    func makeBody(configuration: Configuration) -> some View {
        let pressed = configuration.isPressed && !isDisabled && !isLoading
        let colors = IldsButtonColors.resolve(
            type: type,
            appearance: appearance,
            isPressed: pressed,
            isDisabled: isDisabled,
            isLoading: isLoading
        )
        let primaryOverlay = IldsButtonColors.primaryPressedOverlay(
            type: type,
            appearance: appearance,
            isPressed: pressed,
            isDisabled: isDisabled,
            isLoading: isLoading
        )

        configuration.label
            .foregroundStyle(colors.foreground)
            .background(colors.background)
            .overlay {
                if let primaryOverlay {
                    RoundedRectangle(cornerRadius: ILDSTokens.radiusLarge)
                        .fill(primaryOverlay)
                }
            }
            .overlay {
                if colors.borderWidth > 0, let border = colors.borderColor {
                    RoundedRectangle(cornerRadius: ILDSTokens.radiusLarge)
                        .strokeBorder(border, lineWidth: colors.borderWidth)
                }
            }
            .clipShape(RoundedRectangle(cornerRadius: ILDSTokens.radiusLarge))
            .animation(.easeInOut(duration: 0.15), value: pressed)
    }
}

// MARK: - Metrics

private struct IldsButtonMetrics {
    let padding: EdgeInsets
    let gap: CGFloat
    let minHeight: CGFloat
    let iconSlot: CGFloat
    let font: Font

    init(type: IldsButtonType, size: IldsButtonSize, iconOnly: Bool) {
        if iconOnly && size == .small {
            padding = EdgeInsets(top: ILDSTokens.sp6, leading: ILDSTokens.sp8, bottom: ILDSTokens.sp6, trailing: ILDSTokens.sp8)
        } else if type == .tertiary {
            switch size {
            case .large:
                padding = EdgeInsets(top: ILDSTokens.sp12, leading: 0, bottom: ILDSTokens.sp12, trailing: 0)
            case .medium:
                padding = EdgeInsets(top: ILDSTokens.sp8, leading: 0, bottom: ILDSTokens.sp8, trailing: 0)
            case .small:
                padding = EdgeInsets(top: ILDSTokens.sp6, leading: 0, bottom: ILDSTokens.sp6, trailing: 0)
            }
        } else {
            switch size {
            case .large:
                padding = EdgeInsets(top: ILDSTokens.sp12, leading: ILDSTokens.sp16, bottom: ILDSTokens.sp12, trailing: ILDSTokens.sp16)
            case .medium:
                padding = EdgeInsets(top: ILDSTokens.sp8, leading: ILDSTokens.sp12, bottom: ILDSTokens.sp8, trailing: ILDSTokens.sp12)
            case .small:
                padding = EdgeInsets(top: ILDSTokens.sp6, leading: ILDSTokens.sp12, bottom: ILDSTokens.sp6, trailing: ILDSTokens.sp12)
            }
        }
        switch size {
        case .large, .medium:
            gap = ILDSTokens.sp8
            iconSlot = size == .large ? 24 : 20
            minHeight = size == .large ? 48 : 36
            font = size == .large
                ? .system(size: ILDSTokens.fontSize16, weight: ILDSTokens.fontWeightBold)
                : .system(size: ILDSTokens.fontSize14, weight: ILDSTokens.fontWeightBold)
        case .small:
            gap = ILDSTokens.sp6
            iconSlot = 12
            minHeight = 28
            font = .system(size: ILDSTokens.fontSize12, weight: ILDSTokens.fontWeightBold)
        }
    }
}

// MARK: - Color resolution (mirrors Flutter _resolveColors)

private struct IldsButtonColors {
    let background: Color
    let foreground: Color
    let borderColor: Color?
    let borderWidth: CGFloat

    static func accent(for appearance: IldsButtonAppearance) -> Color {
        appearance == .normal ? ILDSTokens.primaryOrange500 : ILDSTokens.errorRed600
    }

    static func primaryPressedOverlay(
        type: IldsButtonType,
        appearance: IldsButtonAppearance,
        isPressed: Bool,
        isDisabled: Bool,
        isLoading: Bool
    ) -> Color? {
        guard type == .primary, isPressed, !isDisabled, !isLoading else { return nil }
        return appearance == .normal ? ILDSTokens.primaryOrange600 : ILDSTokens.errorRed700
    }

    static func resolve(
        type: IldsButtonType,
        appearance: IldsButtonAppearance,
        isPressed: Bool,
        isDisabled: Bool,
        isLoading: Bool
    ) -> IldsButtonColors {
        let accent = accent(for: appearance)

        if isDisabled {
            switch type {
            case .primary:
                return IldsButtonColors(
                    background: ILDSTokens.neutralCoolgray400,
                    foreground: ILDSTokens.globalWhite000,
                    borderColor: nil,
                    borderWidth: 0
                )
            case .secondary:
                return IldsButtonColors(
                    background: ILDSTokens.neutralCoolgray50,
                    foreground: ILDSTokens.neutralCoolgray400,
                    borderColor: ILDSTokens.neutralCoolgray400,
                    borderWidth: 1
                )
            case .tertiary:
                return IldsButtonColors(
                    background: .clear,
                    foreground: ILDSTokens.neutralCoolgray400,
                    borderColor: nil,
                    borderWidth: 0
                )
            }
        }

        if isLoading {
            switch type {
            case .primary:
                return IldsButtonColors(background: accent, foreground: ILDSTokens.globalWhite000, borderColor: nil, borderWidth: 0)
            case .secondary:
                return IldsButtonColors(background: ILDSTokens.globalWhite000, foreground: accent, borderColor: accent, borderWidth: 1)
            case .tertiary:
                return IldsButtonColors(background: .clear, foreground: accent, borderColor: nil, borderWidth: 0)
            }
        }

        if isPressed {
            switch type {
            case .primary:
                break // overlay handles primary pressed
            case .secondary:
                if appearance == .normal {
                    return IldsButtonColors(
                        background: ILDSTokens.primaryOrange100,
                        foreground: ILDSTokens.primaryOrange600,
                        borderColor: ILDSTokens.primaryOrange600,
                        borderWidth: 1
                    )
                }
                return IldsButtonColors(
                    background: ILDSTokens.errorRed100,
                    foreground: ILDSTokens.errorRed700,
                    borderColor: ILDSTokens.errorRed600,
                    borderWidth: 1
                )
            case .tertiary:
                if appearance == .normal {
                    return IldsButtonColors(background: .clear, foreground: ILDSTokens.primaryOrange600, borderColor: nil, borderWidth: 0)
                }
                return IldsButtonColors(background: .clear, foreground: ILDSTokens.errorRed700, borderColor: nil, borderWidth: 0)
            }
        }

        switch type {
        case .primary:
            return IldsButtonColors(background: accent, foreground: ILDSTokens.globalWhite000, borderColor: nil, borderWidth: 0)
        case .secondary:
            return IldsButtonColors(background: ILDSTokens.globalWhite000, foreground: accent, borderColor: accent, borderWidth: 1)
        case .tertiary:
            return IldsButtonColors(background: .clear, foreground: accent, borderColor: nil, borderWidth: 0)
        }
    }
}

File: android/ilds-design-system/src/main/kotlin/com/icicilombard/ilds/components/IldsButton.kt
package com.icicilombard.ilds.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.interaction.MutableInteractionSource
import androidx.compose.foundation.interaction.collectIsPressedAsState
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.defaultMinSize
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.Immutable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.icicilombard.ilds.tokens.IldsTokens

// Figma component set 13472:2804 — mirrors lib/ilds_button.dart + ios/IldsButton.swift.

enum class IldsButtonType { Primary, Secondary, Tertiary }

enum class IldsButtonSize { Large, Medium, Small }

enum class IldsButtonAppearance { Normal, Destructive }

@Composable
fun IldsButton(
    label: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    type: IldsButtonType = IldsButtonType.Primary,
    size: IldsButtonSize = IldsButtonSize.Large,
    appearance: IldsButtonAppearance = IldsButtonAppearance.Normal,
    isDisabled: Boolean = false,
    isLoading: Boolean = false,
    leading: (@Composable () -> Unit)? = null,
    trailing: (@Composable () -> Unit)? = null,
) {
    IldsButtonSurface(
        onClick = onClick,
        modifier = modifier.semantics { contentDescription = label },
        type = type,
        size = size,
        appearance = appearance,
        isDisabled = isDisabled,
        isLoading = isLoading,
        iconOnly = false,
    ) { colors, metrics ->
        val showTrailingIcon = trailing != null && !isLoading
        Row(
            modifier = Modifier.defaultMinSize(minHeight = metrics.minHeight),
            horizontalArrangement = Arrangement.Center,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            if (leading != null) {
                RowSlot(size = metrics.iconSlot, contentColor = colors.foreground, content = leading)
                Spacer(modifier = Modifier.size(metrics.gap))
            }
            Text(
                text = label,
                style = metrics.textStyle,
                color = colors.foreground,
                maxLines = 1,
            )
            if (isLoading) {
                Spacer(modifier = Modifier.size(metrics.gap))
                CircularProgressIndicator(
                    modifier = Modifier.size(metrics.iconSlot),
                    color = colors.foreground,
                    strokeWidth = metrics.progressStroke,
                )
            } else if (showTrailingIcon) {
                Spacer(modifier = Modifier.size(metrics.gap))
                RowSlot(size = metrics.iconSlot, contentColor = colors.foreground, content = trailing!!)
            }
        }
    }
}

/** Icon-only button (Figma 13472:2810 L, 13472:3718 S). */
@Composable
fun IldsIconButton(
    onClick: () -> Unit,
    semanticLabel: String,
    icon: @Composable () -> Unit,
    modifier: Modifier = Modifier,
    type: IldsButtonType = IldsButtonType.Primary,
    size: IldsButtonSize = IldsButtonSize.Large,
    appearance: IldsButtonAppearance = IldsButtonAppearance.Normal,
    isDisabled: Boolean = false,
    isLoading: Boolean = false,
) {
    IldsButtonSurface(
        onClick = onClick,
        modifier = modifier.semantics { contentDescription = semanticLabel },
        type = type,
        size = size,
        appearance = appearance,
        isDisabled = isDisabled,
        isLoading = isLoading,
        iconOnly = true,
    ) { colors, metrics ->
        Row(
            modifier = Modifier.defaultMinSize(minHeight = metrics.minHeight),
            horizontalArrangement = Arrangement.Center,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            if (isLoading) {
                CircularProgressIndicator(
                    modifier = Modifier.size(metrics.iconSlot),
                    color = colors.foreground,
                    strokeWidth = metrics.progressStroke,
                )
            } else {
                RowSlot(size = metrics.iconSlot, contentColor = colors.foreground, content = icon)
            }
        }
    }
}

@Composable
private fun IldsButtonSurface(
    onClick: () -> Unit,
    modifier: Modifier,
    type: IldsButtonType,
    size: IldsButtonSize,
    appearance: IldsButtonAppearance,
    isDisabled: Boolean,
    isLoading: Boolean,
    iconOnly: Boolean,
    content: @Composable (IldsButtonColors, IldsButtonMetrics) -> Unit,
) {
    val interactionSource = remember { MutableInteractionSource() }
    val isPressed by interactionSource.collectIsPressedAsState()
    val interactive = !isDisabled && !isLoading
    val pressed = isPressed && interactive
    val colors = remember(type, appearance, pressed, isDisabled, isLoading) {
        IldsButtonColors.resolve(type, appearance, pressed, isDisabled, isLoading)
    }
    val metrics = remember(type, size, iconOnly) { IldsButtonMetrics(type, size, iconOnly) }
    val primaryOverlay = IldsButtonColors.primaryPressedOverlay(type, appearance, pressed, isDisabled, isLoading)

    val surfaceColor = if (type == IldsButtonType.Primary && primaryOverlay != null) {
        primaryOverlay
    } else {
        colors.background
    }

    Surface(
        onClick = onClick,
        modifier = modifier,
        enabled = interactive,
        interactionSource = interactionSource,
        shape = RoundedCornerShape(IldsTokens.radiusLarge),
        color = surfaceColor,
        contentColor = colors.foreground,
        border = if (colors.borderWidth > 0.dp && colors.borderColor != null) {
            BorderStroke(colors.borderWidth, colors.borderColor)
        } else {
            null
        },
    ) {
        androidx.compose.foundation.layout.Box(
            modifier = Modifier.padding(metrics.padding),
            contentAlignment = Alignment.Center,
        ) {
            content(colors, metrics)
        }
    }
}

@Composable
private fun RowSlot(
    size: Dp,
    contentColor: Color,
    content: @Composable () -> Unit,
) {
    androidx.compose.foundation.layout.Box(
        modifier = Modifier.size(size),
        contentAlignment = Alignment.Center,
    ) {
        androidx.compose.runtime.CompositionLocalProvider(
            androidx.compose.material3.LocalContentColor provides contentColor,
        ) {
            content()
        }
    }
}

@Immutable
private data class IldsButtonColors(
    val background: Color,
    val foreground: Color,
    val borderColor: Color?,
    val borderWidth: Dp,
) {
    companion object {
        private fun accent(appearance: IldsButtonAppearance): Color =
            if (appearance == IldsButtonAppearance.Normal) IldsTokens.primaryOrange500
            else IldsTokens.errorRed600

        fun primaryPressedOverlay(
            type: IldsButtonType,
            appearance: IldsButtonAppearance,
            isPressed: Boolean,
            isDisabled: Boolean,
            isLoading: Boolean,
        ): Color? {
            if (type != IldsButtonType.Primary || !isPressed || isDisabled || isLoading) return null
            return if (appearance == IldsButtonAppearance.Normal) IldsTokens.primaryOrange600
            else IldsTokens.errorRed700
        }

        fun resolve(
            type: IldsButtonType,
            appearance: IldsButtonAppearance,
            isPressed: Boolean,
            isDisabled: Boolean,
            isLoading: Boolean,
        ): IldsButtonColors {
            val accentColor = accent(appearance)

            if (isDisabled) {
                return when (type) {
                    IldsButtonType.Primary -> IldsButtonColors(
                        IldsTokens.neutralCoolgray400,
                        IldsTokens.globalWhite000,
                        null,
                        0.dp,
                    )
                    IldsButtonType.Secondary -> IldsButtonColors(
                        IldsTokens.neutralCoolgray50,
                        IldsTokens.neutralCoolgray400,
                        IldsTokens.neutralCoolgray400,
                        1.dp,
                    )
                    IldsButtonType.Tertiary -> IldsButtonColors(
                        Color.Transparent,
                        IldsTokens.neutralCoolgray400,
                        null,
                        0.dp,
                    )
                }
            }

            if (isLoading) {
                return when (type) {
                    IldsButtonType.Primary -> IldsButtonColors(accentColor, IldsTokens.globalWhite000, null, 0.dp)
                    IldsButtonType.Secondary -> IldsButtonColors(IldsTokens.globalWhite000, accentColor, accentColor, 1.dp)
                    IldsButtonType.Tertiary -> IldsButtonColors(Color.Transparent, accentColor, null, 0.dp)
                }
            }

            if (isPressed) {
                when (type) {
                    IldsButtonType.Primary -> { /* overlay handles */ }
                    IldsButtonType.Secondary -> {
                        return if (appearance == IldsButtonAppearance.Normal) {
                            IldsButtonColors(
                                IldsTokens.primaryOrange100,
                                IldsTokens.primaryOrange600,
                                IldsTokens.primaryOrange600,
                                1.dp,
                            )
                        } else {
                            IldsButtonColors(
                                IldsTokens.errorRed100,
                                IldsTokens.errorRed700,
                                IldsTokens.errorRed600,
                                1.dp,
                            )
                        }
                    }
                    IldsButtonType.Tertiary -> {
                        return if (appearance == IldsButtonAppearance.Normal) {
                            IldsButtonColors(Color.Transparent, IldsTokens.primaryOrange600, null, 0.dp)
                        } else {
                            IldsButtonColors(Color.Transparent, IldsTokens.errorRed700, null, 0.dp)
                        }
                    }
                }
            }

            return when (type) {
                IldsButtonType.Primary -> IldsButtonColors(accentColor, IldsTokens.globalWhite000, null, 0.dp)
                IldsButtonType.Secondary -> IldsButtonColors(IldsTokens.globalWhite000, accentColor, accentColor, 1.dp)
                IldsButtonType.Tertiary -> IldsButtonColors(Color.Transparent, accentColor, null, 0.dp)
            }
        }
    }
}

@Immutable
private data class IldsButtonMetrics(
    val padding: PaddingValues,
    val gap: Dp,
    val minHeight: Dp,
    val iconSlot: Dp,
    val textStyle: TextStyle,
    val progressStroke: Dp,
) {
    constructor(type: IldsButtonType, size: IldsButtonSize, iconOnly: Boolean) : this(
        padding = buttonPadding(type, size, iconOnly),
        gap = when (size) {
            IldsButtonSize.Large, IldsButtonSize.Medium -> IldsTokens.sp8
            IldsButtonSize.Small -> IldsTokens.sp6
        },
        minHeight = when (size) {
            IldsButtonSize.Large -> 48.dp
            IldsButtonSize.Medium -> 36.dp
            IldsButtonSize.Small -> 28.dp
        },
        iconSlot = when (size) {
            IldsButtonSize.Large -> 24.dp
            IldsButtonSize.Medium -> 20.dp
            IldsButtonSize.Small -> 12.dp
        },
        textStyle = when (size) {
            IldsButtonSize.Large -> TextStyle(
                fontSize = IldsTokens.fontSize16,
                lineHeight = 20.sp,
                fontWeight = IldsTokens.fontWeightBold,
            )
            IldsButtonSize.Medium -> TextStyle(
                fontSize = IldsTokens.fontSize14,
                lineHeight = 16.sp,
                fontWeight = IldsTokens.fontWeightBold,
            )
            IldsButtonSize.Small -> TextStyle(
                fontSize = IldsTokens.fontSize12,
                lineHeight = 16.sp,
                fontWeight = IldsTokens.fontWeightBold,
            )
        },
        progressStroke = when (size) {
            IldsButtonSize.Large -> 2.5.dp
            IldsButtonSize.Medium -> 2.25.dp
            IldsButtonSize.Small -> 2.dp
        },
    )

}

private fun buttonPadding(
    type: IldsButtonType,
    size: IldsButtonSize,
    iconOnly: Boolean,
): PaddingValues {
    if (iconOnly && size == IldsButtonSize.Small) {
        return PaddingValues(horizontal = IldsTokens.sp8, vertical = IldsTokens.sp6)
    }
    if (type == IldsButtonType.Tertiary) {
        return when (size) {
            IldsButtonSize.Large -> PaddingValues(vertical = IldsTokens.sp12)
            IldsButtonSize.Medium -> PaddingValues(vertical = IldsTokens.sp8)
            IldsButtonSize.Small -> PaddingValues(vertical = IldsTokens.sp6)
        }
    }
    return when (size) {
        IldsButtonSize.Large -> PaddingValues(horizontal = IldsTokens.sp16, vertical = IldsTokens.sp12)
        IldsButtonSize.Medium -> PaddingValues(horizontal = IldsTokens.sp12, vertical = IldsTokens.sp8)
        IldsButtonSize.Small -> PaddingValues(horizontal = IldsTokens.sp12, vertical = IldsTokens.sp6)
    }
}
